examples/04_fuzzy_joining.py

```diff
@@ -406,7 +407,9 @@
     remainder="drop",
 )
 
-pipeline = make_pipeline(joiner, encoder, HistGradientBoostingRegressor())
+pipeline = Pipeline(
```

Comment:
sorry the example was outdated after the merge of the SelectCols pr, should be a bit better now

Reply:
after merging `main` this PR makes very few changes to the example. So I suggest we discuss and apply any other changes in this example in #802  (or a later PR) to avoid duplicating the review work and having  too many merge conflicts. WDYT?